source/bringauto/modules/ModuleManagerLibraryHandler.cpp

```diff
@@ -113,4 +120,17 @@ void ModuleManagerLibraryHandler::deallocate(struct buffer *buffer){
 	deallocate_(buffer);
 }
 
-}
\ No newline at end of file
+bringauto::modules::Buffer ModuleManagerLibraryHandler::constructBuffer(std::size_t size) {
+	struct ::buffer buff;
+	buff.size_in_bytes = size;
+	if(allocate(&buff, size) != OK) {
+		throw std::bad_alloc();
+	}
+	return { buff, deallocate_ };
+}
+
+bringauto::modules::Buffer ModuleManagerLibraryHandler::constructBufferByTakeOwnership(struct ::buffer &buffer) {
+	return { buffer, deallocate_ };
```

Comment:
I think about checking the buffer itself. Maybe the check if the buffer is not empty (data != nullptr) should be added? To improve consistency and  avoidsegfault by a module impl. bug?

Reply:
Added check if buffer is allocated